### PR TITLE
[Merged by Bors] - feat: When the symmetric difference has finite measure

### PR DESCRIPTION
**Pull request by EtienneC30**
Add the following lemma: is `s` and `t` are two measurable sets such that `s` and `s ∆ t` have finite measures, then so does `t`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
Can you rather show that if `s ∆ t` has finite measure, then `s` has finite measure iff `t` has finite measure?

**Comment by EtienneC30** [comment]
Done! I also added the two implications.

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 1]
```diff
@@ -256,6 +256,30 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
 
 #align measure_theory.le_measure_diff MeasureTheory.le_measure_diff
 
+open scoped symmDiff in
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:259]: Can you just `open scoped symmDiff` at the top of the file?

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 2]
```diff
@@ -256,6 +256,30 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
...
+    → μ (u ∆ v) ≠ ∞ → μ u = ∞ → μ v = ∞
+    from ⟨h s t hs ht hμst, h t s ht hs (symmDiff_comm s t ▸ hμst)⟩
+  intro u v hu hv hμuv hμu
+  by_contra! hμv
+  apply hμuv
+  rw [Set.symmDiff_def, measure_union₀ (hv.diff hu) disjoint_sdiff_sdiff.aedisjoint, add_eq_top]
+  left
+  rw [eq_top_iff]
+  calc
+    ∞ = μ u - μ v :=(WithTop.sub_eq_top_iff.2 ⟨hμu, hμv⟩).symm
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:274]:
```suggestion
    ∞ = μ u - μ v := (WithTop.sub_eq_top_iff.2 ⟨hμu, hμv⟩).symm
```

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 3]
```diff
@@ -256,6 +254,28 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
 
 #align measure_theory.le_measure_diff MeasureTheory.le_measure_diff
 
+/-- If the measure of the symmetric difference of two null measurable sets is finite,
+then one has infinite measure if and only if the other does. -/
+theorem Ne.measure_eq_top_iff_of_symmDiff (hs : NullMeasurableSet s μ) (ht : NullMeasurableSet t μ)
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:259]: I don't think the `Ne` brings anything here. Same below.
```suggestion
theorem measure_eq_top_iff_of_symmDiff (hs : NullMeasurableSet s μ) (ht : NullMeasurableSet t μ)
```

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 4]
```diff
@@ -256,6 +254,28 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
 
 #align measure_theory.le_measure_diff MeasureTheory.le_measure_diff
 
+/-- If the measure of the symmetric difference of two null measurable sets is finite,
+then one has infinite measure if and only if the other does. -/
+theorem measure_eq_top_iff_of_symmDiff (hs : NullMeasurableSet s μ) (ht : NullMeasurableSet t μ)
+    (hμst : μ (s ∆ t) ≠ ∞) : μ s = ∞ ↔ μ t = ∞ := by
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:260]: I'm wondering whether you can get rid of the measurability assumption by using `measure_eq_iInf`/`measure_eq_iInf'`. Would you be willing to try?

Comment (EtienneC30) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean:260]: Ok I'll try.

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 5]
```diff
@@ -256,6 +254,25 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
 
 #align measure_theory.le_measure_diff MeasureTheory.le_measure_diff
 
+/-- If the measure of the symmetric difference of two sets is finite,
+then one has infinite measure if and only if the other does. -/
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:258]: I am afraid you are French.
```suggestion
then one has infinite measure if and only if the other one does. -/
```

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 6]
```diff
@@ -256,6 +254,25 @@ theorem le_measure_diff : μ s₁ - μ s₂ ≤ μ (s₁ \ s₂) :=
...
+  by_contra! hμv
+  apply hμuv
+  rw [Set.symmDiff_def, eq_top_iff]
+  calc
+    ∞ = μ u - μ v := (WithTop.sub_eq_top_iff.2 ⟨hμu, hμv⟩).symm
+    _ ≤ μ (u \ v) := le_measure_diff
+    _ ≤ μ (u \ v ∪ v \ u) := measure_mono <| subset_union_left ..
+
+/-- If the measure of the symmetric difference of two sets is finite,
+then one has finite measure if and only if the other does. -/
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:272]:
```suggestion
then one has finite measure if and only if the other one does. -/
```

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean (hunk 5 above)
Comment (EtienneC30) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean:258]: Indeed...

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**